cloud/misc/ovirt_storage.py

```diff
@@ -0,0 +1,432 @@
+    """
+    Attach specified disk to VM
+    Parameters:
+        disk - ovirtsdk.infrastructure.brokers.Disk object to attach
+        vm - ovirtsdk.infrastructure.brokers.VM object to attach disk to
+    Returns:
+        0 - No change
+        1 - Change
+    """
+    vm_disk = vm.disks.get(id=disk.id)
+    # Check if disk is already attached to the VM
+    if vm_disk:
+        # Disk is already attached so lets activate it
+        if activate:
+            return _activate(vm_disk)
+    else:
+        if not MODULE_CHECK_MODE:
+            vm_disk = vm.disks.add(params.Disk(id=disk.id, active=activate))
+        return 1
+    return 0
+
+
+def _detach(vm, disk):
+    """
+    Detach specified disk from VM
+    Parameters:
+        disk - ovirtsdk.infrastructure.brokers.VMDisk object to detach
+        vm - ovirtsdk.infrastructure.brokers.VM object to detach disk from
+    Returns:
+        0 - No change
+        1 - Change
+    """
+    if not MODULE_CHECK_MODE:
+        disk.delete(action=params.Action(detach=True))
+    return 1
+
+
+def attach_disk(api, vm_name, disk_alias, activate=True):
+    """
+    Attach disk_alias to vm_name then activate it by default
+    Parameters:
+      vm_name - string of VM to attach disk to
+      disk_alias - string of disk to attach to VM
+    """
+    try:
+        vm = api.vms.get(name=vm_name)
+        disks = api.disks.list(alias=disk_alias)
+        ret = 0
+        if not disks:
+            raise Exception("No disks with name " + disk_alias + " found")
+        for disk in disks.__iter__():
+            ret += _attach(vm, disk, activate)
+        return ret
+    except Exception as e:
+        raise Exception("Error while attaching disk")
+
+
+def detach_disk(api, vm_name, disk_alias, detach=True):
+    """
+    Deactivate disk_alias and by default detach it from vm_name
+    """
+    try:
+        vm = api.vms.get(name=vm_name)
+        disks = vm.disks.list(alias=disk_alias)
+        ret = 0
+        if not disks:
+            raise Exception("No disks with name " + disk_alias + " are attached to " + vm_name)
+        for disk in disks.__iter__():
+            ret += _deactivate(disk)
+            if detach:
+                ret += _detach(vm, disk)
+        return ret
+    except Exception as e:
+        raise Exception("Exception while detaching disk " + str(e))
+
+
+def create_disk(api, vm_name, disk_alias, disk_size_gb, disk_alloc, disk_iface):
+    """Create a new disk with specified name and size. Attach to vm_name"""
+    try:
+        vm = api.vms.get(name=vm_name)
+        disks = vm.disks.list(alias=disk_alias)
+        if disks:
+            return 0
+        if not MODULE_CHECK_MODE:
+            size = int(disk_size_gb) * 1024 * 1024 * 1024
+            disk_params = params.Disk()
+            disk_params.set_wipe_after_delete(True)
+            disk_params.set_active(True)
+            disk_params.set_alias(disk_alias)
+            disk_params.set_size(size)
```

Comment:
The `size` attribute of disks has been deprecated, and replaced by `provisioned_size`. It is just a name change, consider using the new name.